In [10]:
# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [11]:
# Importing prepared dataset
import pickle
input = open('data_klast.pkl', 'rb')
obj = pickle.load(input)
input.close()
X_train = obj["X_train"]
X_test = obj["X_test"]
y_train = obj["y_train"]
y_test = obj["y_test"]

In [12]:
# Fitting SVM to the Training set (4 variables)
from sklearn.svm import SVC
svm = SVC(kernel = 'rbf', random_state = 10).fit(X_train, y_train)

In [13]:
# Predicting the Test set results
y_pred = svm.predict(X_test)
svm.score(X_test,y_test)

0.8080357142857143

In [14]:
# Making the Confusion Matrix
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)
print(cm)

[[321  19]
 [ 67  41]]


In [15]:
svm = SVC(kernel = 'linear', random_state = 10).fit(X_train, y_train)

In [16]:
y_pred = svm.predict(X_test)
svm.score(X_test,y_test)

0.8125

In [17]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)
print(cm)

[[320  20]
 [ 64  44]]


In [18]:
svm = SVC(kernel = 'poly', random_state = 10).fit(X_train, y_train)

In [19]:
y_pred = svm.predict(X_test)
svm.score(X_test,y_test)

0.8035714285714286

In [20]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)
print(cm)

[[330  10]
 [ 78  30]]


In [21]:
svm = SVC(kernel = 'sigmoid', random_state = 10).fit(X_train, y_train)

In [22]:
y_pred = svm.predict(X_test)
svm.score(X_test,y_test)

0.6808035714285714

In [23]:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)
print(cm)

[[256  84]
 [ 59  49]]


Все модели также получили приблизительно одинаковые результаты, кроме модели постоенной на сигмоиде - общие результаты хуже, чем у всех предыдущтх моделях.

Наверное, лучшей можно выделить модель, которая постороена на линейном ядре. 

Точность = 81%
Чувствительность = 41%
Специфичность = 94%